src/promptflow-core/promptflow/_utils/multimedia_utils.py

```diff
@@ -2,278 +2,497 @@
-        if resource == "path":
-            return _create_image_from_file(Path(v), mime_type=f"image/{format}")
-        elif resource == "base64":
-            if _is_base64(v):
-                return _create_image_from_base64(v, mime_type=f"image/{format}")
+    @staticmethod
+    def create_from_yaml(flow_file: Path, working_dir: Optional[Path] = None):
+        if flow_file and Path(flow_file).suffix.lower() in [".yaml", ".yml"]:
+            flow_file = working_dir / flow_file if working_dir else flow_file
+            with open(flow_file, "r", encoding="utf-8") as fin:
+                flow_dag = load_yaml(fin)
+            message_format_type = flow_dag.get("message_format", MessageFormatType.BASIC)
+            return MultimediaProcessor.create(message_format_type)
+        return BasicMultimediaProcessor()
+
+    def create_image(self, value: any):
+        if isinstance(value, PFBytes):
+            return value
+        elif isinstance(value, dict):
+            if self.is_multimedia_dict(value):
+                return self._create_image_from_dict(value)
             else:
                 raise InvalidImageInput(
-                    message_format=f"Invalid base64 image: {v}.",
+                    message_format="Invalid image input format. The image input should be a dictionary like: "
+                    "{{data:image/<image_type>;[path|base64|url]: <image_data>}}.",
                     target=ErrorTarget.EXECUTOR,
                 )
-        elif resource == "url":
-            return _create_image_from_url(v, mime_type=f"image/{format}")
+        elif isinstance(value, str):
+            if not value:
+                raise InvalidImageInput(
+                    message_format="The image input should not be empty.", target=ErrorTarget.EXECUTOR
+                )
+            return ImageProcessor.create_image_from_string(value)
         else:
             raise InvalidImageInput(
-                message_format=f"Unsupported image resource: {resource}. "
-                "Supported Resources are [path, base64, url].",
+                message_format=f"Unsupported image input type: {type(value)}. "
+                "The image inputs should be a string or a dictionary.",
                 target=ErrorTarget.EXECUTOR,
             )
 
+    def _save_image_to_file(
+        self, image: Image, file_name: str, folder_path: Path, relative_path: Path = None, use_absolute_path=False
+    ):
+        ext = ImageProcessor.get_extension_from_mime_type(image._mime_type)
+        file_name = f"{file_name}.{ext}" if ext else file_name
+        image_path = (relative_path / file_name).as_posix() if relative_path else file_name
+        if use_absolute_path:
+            image_path = Path(folder_path / image_path).resolve().as_posix()
+        image_reference = self._generate_image_file_reference(image, image_path)
+        path = folder_path / relative_path if relative_path else folder_path
+        os.makedirs(path, exist_ok=True)
```

Comment:
The pathlib convention of this line should be:
```python
path.mkdir(parents=True, exist_ok=True)
```

Refer to https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module for details.

Reply:
This code is historical, let's keep it that way.